File: src/main/java/com/avs/filmoteca/data/ws/BackendInterface.java
package com.avs.filmoteca.data.ws;

import java.util.List;

import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import rx.Observable;

public interface BackendInterface {
	
	@POST("movies")
	Observable<Void> getUpdateMoviesObservable(@Body List<String> movies);

	@GET("movies")
	Observable<List<String>> getStoredMoviesObservable();
	
	@GET("registrationId")
	Observable<List<String>> getRegistrationIdsObservable();

}

File: src/main/java/com/avs/filmoteca/data/ws/BackendInterface.kt
package com.avs.filmoteca.data.ws

import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query
import rx.Observable

interface BackendInterface {

    @GET("movies")
    fun getStoredMoviesObservable(@Query("region") region: String): Observable<List<String>>

    @GET("registrationId")
    fun getRegistrationIdsObservable(@Query("region") region: String): Observable<List<String>>

    @POST("movies")
    fun getUpdateMoviesObservable(@Body movies: List<String>, @Query("region") region: String): Observable<Void>

}
